### PR TITLE
CTX-5504: Object Detection - Fixed memory leak caused by matplotlib plots

### PR DESCRIPTION
**Pull request by VukManojlovic**

### CODE REVIEW

File: tasks/object-detection-yolov8/src/predict.py [hunk 1]
```diff
@@ -13,27 +13,42 @@ def classByLabelId(labelId: int, classes: ImageDatasetClasses) -> Optional[Image
...
+            ))
+
+    plt.savefig(savePath)
+
+
+def run(model: YOLO, dataset: ComputerVisionDataset, resultPath: Path, batchSize: int) -> None:
+    sampleBatch: list[Path] = []
+    for i, sample in enumerate(dataset.samples):
+        if any([len(segmentation) < 6 for instance in sample.load().annotation.instances
+                for segmentation in instance.segmentations]):
```
Comment (dule1322) [inline tasks/object-detection-yolov8/src/predict.py:44]: Let's extract this into a separate function to simplify

File: tasks/object-detection-yolov8/src/predict.py [hunk 2]
```diff
@@ -13,27 +13,42 @@ def classByLabelId(labelId: int, classes: ImageDatasetClasses) -> Optional[Image
...
+                edgecolor = clazz.color,
+                facecolor = "none"
+            ))
+
+    plt.savefig(savePath)
+
+
+def run(model: YOLO, dataset: ComputerVisionDataset, resultPath: Path, batchSize: int) -> None:
+    sampleBatch: list[Path] = []
+    for i, sample in enumerate(dataset.samples):
```
Comment (dule1322) [inline tasks/object-detection-yolov8/src/predict.py:42]: Isn't there a better way to do batching? ex. `for i in range(0, len(dataset.samples), batchSize)`, and then for the remained it is possible to get it like `dataset.count % batchSize`.

**Review by dule1322 — CHANGES_REQUESTED** [review]

File: tasks/object-detection-yolov8/src/predict.py [hunk 3]
```diff
@@ -13,27 +13,46 @@ def classByLabelId(labelId: int, classes: ImageDatasetClasses) -> Optional[Image
...
+    return True
+
+
+def run(model: YOLO, dataset: ComputerVisionDataset, resultPath: Path, batchSize: int) -> None:
+    for i in range(0, len(dataset.samples), batchSize):
+        print(i)
+        print(dataset.count - i)
+        print(i + batchSize)
+        if (dataset.count - i) < batchSize:
+            batchSize = (dataset.count - i)
```
Comment (dule1322) [inline tasks/object-detection-yolov8/src/predict.py:54]: Why is there a need for this condition?

Comment (VukManojlovic) [reply tasks/object-detection-yolov8/src/predict.py:51]: To adjust the final batch to the size of the remainder

Comment (dule1322) [reply tasks/object-detection-yolov8/src/predict.py:51]: Let's do processing of remainder outside of the for-loop. It is easy to get the reminder with modulo operator

**Pull request merged**